File: server/src/main/java/com/bomberman/entities/Wall.java
package com.bomberman.entities;

import com.bomberman.constants.MapConst;

public class Wall extends MapObject {

  public Wall(final Position position) {
    super(position);
  }

  @Override
  public int getTextureWidth() {
    return MapConst.TEXTURE_SIZE;
  }

  @Override
  public int getTextureHeight() {
    return MapConst.TEXTURE_SIZE;
  }
}

File: client/src/entities/mapobject/Wall.ts
import {MapObject} from './MapObject';
import {Adapter} from "../Adapter";
import {Position, PositionAdapter} from "../Position";

export class Wall extends MapObject {

    constructor(position: Position) {
        super(position);
    }
}

export class WallAdapter implements Adapter<Wall> {
    adapt(item): Wall {
        if (!item) {
            return null;
        }
        const positionAdapter = new PositionAdapter();
        let position = positionAdapter.adapt(item.position)
        return new Wall(position);
    }
}
